File: MAP/Lab 2/ToyInterpreter/src/model/expressions/VarExp.java
package model.expressions;

import model.adts.MyIDictionary;

public class VarExp implements IExp {
    private String expr;

    public VarExp(String expr) {
        this.expr = expr;
    }

    @Override
    public int eval(MyIDictionary<String, Integer> d) throws Exception {
       if(d.find(expr))
           return d.get(expr);

       throw new Exception("Variable not found");
    }

    @Override
    public String toString() {
        return expr;
    }
}

File: MAP/Lab10/ConsoleApp1/ConsoleApp1/Model/expressions/VarExp.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToyInterpreter.Model.expressions
{
    class VarExp: IExp
    {
        private String expr;

        public VarExp(String expr)
        {
            this.expr = expr;
        }

        public int eval(Dictionary<String, Int64> d)
        {
           if(d.ContainsKey(expr))
               return (int) d[key: expr];

           throw new Exception("Variable not found");
        }
  
        public override String ToString()
        {
            return this.expr.ToString();
        }
    }
}
